saleor/graphql/payment/tests/mutations/test_transaction_request_action.py

```diff
@@ -1308,3 +1322,619 @@ def test_transaction_request_cancel_sets_user_to_request_event(
+    transaction_item_generator,
+    permission_group_handle_payments,
+    site_settings,
+):
+    # Given
+    mocked_is_active.return_value = False
+
+    assert site_settings.refund_reason_reference_type is None
+
+    transaction_request_webhook.events.create(
+        event_type=WebhookEventSyncType.TRANSACTION_REFUND_REQUESTED
+    )
+    transaction = transaction_item_generator(
+        checkout_id=checkout.pk,
+        charged_value=Decimal(10),
+        app=transaction_request_webhook.app,
+    )
+
+    variables = {
+        "id": graphene.Node.to_global_id("TransactionItem", transaction.token),
+        "action_type": TransactionActionEnum.REFUND.name,
+        "refund_reason": "Product was damaged during shipping",
+    }
+    staff_api_client.user.groups.add(permission_group_handle_payments)
+
+    # When
+    response = staff_api_client.post_graphql(
+        MUTATION_TRANSACTION_REQUEST_ACTION,
+        variables,
+    )
+
+    # Then
+    content = get_graphql_content(response)
+    data = content["data"]["transactionRequestAction"]
+    errors = data["errors"]
+    assert not errors
+
+    request_event = TransactionEvent.objects.filter(
+        type=TransactionEventType.REFUND_REQUEST,
+    ).first()
+
+    assert request_event
+    assert request_event.user == staff_api_client.user
+    assert request_event.message == "Product was damaged during shipping"
+    assert request_event.reason_reference is None
+
+
+@patch("saleor.plugins.manager.PluginsManager.is_event_active_for_any_plugin")
+@patch("saleor.plugins.manager.PluginsManager.transaction_charge_requested")
+def test_transaction_request_charge_ignores_reason_and_reference(
+    mocked_payment_action_request,
+    mocked_is_active,
+    checkout,
+    staff_api_client,
+    permission_manage_payments,
+    transaction_request_webhook,
+    transaction_item_generator,
+    permission_group_handle_payments,
+    site_settings,
+):
+    # Given
+    mocked_is_active.return_value = False
+
+    page_type = PageType.objects.create(name="Refund Reasons", slug="refund-reasons")
+    page = Page.objects.create(
+        slug="damaged-product",
+        title="Damaged Product",
+        page_type=page_type,
+        is_published=True,
+    )
+
+    site_settings.refund_reason_reference_type = page_type
+    site_settings.save()
+
+    transaction_request_webhook.events.create(
+        event_type=WebhookEventSyncType.TRANSACTION_CHARGE_REQUESTED
+    )
+    transaction = transaction_item_generator(
+        checkout_id=checkout.pk,
+        authorized_value=Decimal(10),
+        app=transaction_request_webhook.app,
+    )
+
+    page_id = to_global_id_or_none(page)
+    variables = {
+        "id": graphene.Node.to_global_id("TransactionItem", transaction.token),
+        "action_type": TransactionActionEnum.CHARGE.name,
+        "refund_reason": "Some reason that should be ignored",
+        "refund_reason_reference": page_id,  # Should be ignored for charge actions
```

Comment:
Is that correct? Do we want to ignore the values?  Your implementation suggests that we should raise a ValidationError:
<img width="667" height="290" alt="image" src="https://github.com/user-attachments/assets/88396dcc-9964-4b2a-b390-8383d1a76352" />

IMHO this test should `assert errors` instead of `assert not errors`.